modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptInjectionPhase.java

```diff
@@ -311,17 +311,11 @@ protected static void injectSandboxExceptions(FunctionNode functionNode) {
 
             CatchNode catchNode = new CatchNode();
             catchNode.setLocation(internalLocation);
+            catchNode.setExceptionType(PainlessExplainError.class);
+            catchNode.setSymbol("#painlessExplainError");
 
             tryNode.addCatchNode(catchNode);
 
-            DeclarationNode declarationNode = new DeclarationNode();
```

Comment:
Let's chat about this.  Is it due to `SDeclaration` removal?

Reply:
`SDeclaration` was embedded in the `SCatch`, `SCatch` can check the type itself.